Cloudflare Gateway DNS Preset

grp-db opened the pull request:
Preset for [Cloudflare](https://developers.cloudflare.com/logs/): [Gateway DNS](https://developers.cloudflare.com/logs/logpush/logpush-job/datasets/account/gateway_dns/)

grp-db committed:
Cloudflare Gateway DNS Preset

alexott reviewed (commented):
You need to register a new data source in `presets/index.yaml`

grp-db committed:
Cloudflare Gateway DNS Preset

1. add metadata.version
2. add src&dst_{endpoint.location.continent&country}
3. add query.packet_uid

grp-db committed:
Cloudflare Gateway DNS Preset

1. augment OCSF schema
2. add metadata tags
3. add new cols (action, disp., obs., policy info, etc.)
4. add source and sourcetype to brz

grp-db committed:
Cloudflare Gateway DNS

1. change brz to variant data type
2. add meta tags
3. remove temp fields and use try_get_variant
4. handle JSON encoded strings (Rdata, resourcerecordsJSON, resourcerecords)

alexott reviewed (commented):
looks good to me, just small code

grp-db commented on presets/cloudflare/gateway_dns/preset.yaml:
```diff
@@ -0,0 +1,525 @@
+        - name: ResolverPolicyName
+          expr: try_variant_get(data, '$.ResolverPolicyName', 'STRING')
+        - name: ResourceRecords
+          expr: from_json(try_variant_get(data, '$.ResourceRecords', 'STRING'), 'ARRAY<MAP<STRING,STRING>>')
```
not really sure although the [docs](https://developers.cloudflare.com/logs/logpush/logpush-job/datasets/account/gateway_dns/#resourcerecords) have it as an `array<object>` data type

grp-db committed:
Cloudflare Gateway DNS Preset

1. change to element_at()

alexott commented:
ported to content packs
